src/lib/model/InboundTransfersModel.js

```diff
@@ -113,6 +117,38 @@ class InboundTransfersModel {
         }
     }
 
+    /**
+     * Queries details of a transfer
+     */
+    async getTransfer(transferId, sourceFspId) {
```

Comment:
@kirgene Incoming GET /transfer requests should not be sourced from the cache but should call out to the DFSP backend with a GET request to the transfers endpoint. (same endpoint as the POST incoming transfer endpoint the DFSP has to implement). It is possible that the DFSP backend will have changed the state of the transfer outside the cache lifetime, or that a request will simply be for a transfer not currently in the cache. It is not safe to assume the cache will have every transfer the DFSP participated in since the beginning of time.

The response from the backend should include all required elements to respond to the request. The call to the DFSP backend does not need to match mojaloop API data models but should follow the existing backend API data model. The response to the switch must be in mojaloop API model.